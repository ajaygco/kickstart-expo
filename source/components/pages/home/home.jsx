// Modules
import React from "react";
import { SafeAreaView } from "react-native-safe-area-context";
import { Text, View } from "react-native";

// Shared
import { Link } from "components/shared/link.shared";

// Component: Presentation
export const HomeUi = () => {
  return (
    <SafeAreaView className="w-full h-full bg-white flex flex-col items-center justify-center">
      <View className="mb-12">
        <Text allowFontScaling={false} className="text-4xl">
          👋
        </Text>
      </View>

      <View className="mb-2">
        <Text
          allowFontScaling={false}
          className="text-3xl font-sans font-black"
        >
          Get Started
        </Text>
      </View>

      <View className="w-full px-8 mb-6">
        <Text allowFontScaling={false} className="text-lg text-center">
          This quickstart has React Navigation, NativeWind and Expo Fonts
          pre-configured and comes with lots of useful components.
        </Text>
      </View>

      <View className="w-full px-8">
        <Link to="https://github.com/ajaygco">
          <Text allowFontScaling={false} className="text-lg text-center">
            Made with 🍕 by{" "}
            <Text allowFontScaling={false} className="text-lg text-blue-500">
              @ajaygco
            </Text>
          </Text>
        </Link>
      </View>
    </SafeAreaView>
  );
};

// Component: Logic
export const Home = () => {
  return <HomeUi />;
};
