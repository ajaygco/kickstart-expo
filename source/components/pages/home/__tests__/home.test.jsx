// bunx jest source/components/pages/home/__tests__/home.test.jsx

import React from "react";
import { render } from "@testing-library/react-native";

import { Home } from "components/pages/home/home";

describe("Home", () => {
  it("should render correctly", () => {
    const wrapper = render(<Home />);

    const titleElement = wrapper.getByText("Get Started");

    expect(titleElement).toBeTruthy();

    expect(wrapper).toMatchSnapshot();
  });
});
