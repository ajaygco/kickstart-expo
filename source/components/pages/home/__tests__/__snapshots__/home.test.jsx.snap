// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "width": "100%",
      },
      {
        "height": "100%",
      },
      {
        "backgroundColor": "#fff",
      },
      {
        "display": "flex",
      },
      {
        "flexDirection": "column",
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "marginBottom": 48,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        [
          {
            "fontSize": 36,
            "lineHeight": 40,
          },
        ]
      }
    >
      👋
    </Text>
  </View>
  <View
    style={
      [
        {
          "marginBottom": 8,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        [
          {
            "fontSize": 30,
            "lineHeight": 36,
          },
          {
            "fontFamily": "Figtree",
          },
          {
            "fontWeight": "900",
          },
        ]
      }
    >
      Get Started
    </Text>
  </View>
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "paddingLeft": 32,
          "paddingRight": 32,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        [
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
          {
            "textAlign": "center",
          },
        ]
      }
    >
      This quickstart has React Navigation, NativeWind and Expo Fonts pre-configured and comes with lots of useful components.
    </Text>
  </View>
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "paddingLeft": 32,
          "paddingRight": 32,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={12}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        allowFontScaling={false}
        style={
          [
            {
              "fontSize": 18,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        Made with 🍕 by
         
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "fontSize": 18,
                "lineHeight": 28,
              },
              {
                "color": "#3b82f6",
              },
            ]
          }
        >
          @ajaygco
        </Text>
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
